### ISSUE TITLE
Magic connector on app that has magic login already integrated

### ISSUE THREAD

**Issue by hozjanmarko** [EveripediaNetwork/wagmi-magic-connector#79]
### Context 
We are working on an application that already uses the magic link SDK and authentication solutions. We are migrating away from ethers and decided to switch it for the rainbowkit/wagmi solution. Since we provide the magic wallet out of the box, it would be natural to offer it also through the raibow kit flow.

### Issue 
For a user to select his magic wallet through raibow kit, he has to login into magic through the rainbow kit integration, even if he is already authenticated with magic on the main app.

### Purposed solution
Provide a separate factory/connector method, where an existing magic instance can be passed along.

Thank you for the effort

**Comment by Royal-lobster** [comment]
Yeah, it makes sense, we can make another export for rainbow kit since most of the users seems to be using wagmi with rainbow kit, tho we have to check out a way to pass in existing magic sdk.
